ext/v8/return-value.h

```diff
@@ -3,88 +3,127 @@
 #define RR_RETURN_VALUE_H
 
 namespace rr {
-  typedef Wrapper<v8::ReturnValue<v8::Value>> ReturnValueWrapper;
 
-  class ReturnValue : public ReturnValueWrapper {
+  class ReturnValue {
   public:
-    ReturnValue(v8::ReturnValue<v8::Value> value) : ReturnValueWrapper(value) {}
-    ReturnValue(VALUE self) : ReturnValueWrapper(self) {}
-
-    static VALUE Set(VALUE self, VALUE handle) {
-      ReturnValue ret(self);
-      Locker lock(ret->GetIsolate());
-      v8::Local<v8::Value> value((Value(handle)));
-      ret->Set(value);
-      return Qnil;
-    }
 
-    static VALUE Set_bool(VALUE self, VALUE value) {
-      ReturnValue ret(self);
-      Locker lock(ret->GetIsolate());
-      ret->Set((bool)Bool(value));
-      return Qnil;
-    }
+    template <class T>
+    class Base : public Wrapper<v8::ReturnValue<T>> {
```

Comment:
Seems like all of the methods should go into the base class. That way, we'll be ready if there is ever a usage of a different type like `ReturnValue<Object>`. We can protect that nobody calls `Set` by just not defining the Ruby method.